main.tf

```diff
@@ -27,26 +145,31 @@ resource "google_container_cluster" "primary" {
 }
 
 module "node-pool" {
-  source       = "./modules/kubernetes_node_pools"
-  region       = "${var.region}"
-  zones        = ["${var.zones}"]
-  project      = "${var.project}"
-  environment  = "${terraform.workspace}"
-  cluster_name = "${google_container_cluster.primary.name}"
-  node_pools   = "${var.node_pools}"
+  source           = "./modules/kubernetes_node_pools"
+  region           = "${coalesce(replace(join("",google_container_cluster.primary.*.zone), "-${var.zones[0]}", ""), replace(join("",google_container_cluster.primary-nat.*.zone), "-${var.zones[0]}", "") , join("",google_container_cluster.primary-regional.*.region), join("",google_container_cluster.primary-regional-nat.*.region))}"
```

Comment:
😮 :)

Reply:
you need to excract it and you dont know which cluster is there :)